CODINGSTYLE.md

```diff
@@ -0,0 +1,541 @@
+There must not be language imports in narrower scopes.
+
+If you import more than 3 or so items from a namespace, use a wildcard import.
+
+Avoid importing mutable collections directly; prefer importing `mutable` and then use `mutable.ListBuffer`.
+
+### ScalaDoc
+
+ScalaDoc comments that fit in one line must be written as
+
+```scala
+/** Returns the maximum of a and b. */
+def max(a: Int, b: Int): Int = ???
+```
+
+Multi-line ScalaDoc comments must use the following style:
+
+```scala
+/** Returns the maximum of a and b.
+ *
+ *  If a > b, returns a. Otherwise returns b.
+ */
+def max(a: Int, b: Int): Int = ???
+```
+
+### Non-ScalaDoc comments
+
+Normal comments fitting on one-line should use `//`.
+A comment that does not fit on one line should use the multi-line comment syntax and follow this style:
+
+```scala
+/* This complicated algorithm computes the maximum of two integer values a
+ * and b. If a > b, it computes a, otherwise it computes b.
+ */
+```
+
+
+## Class declaration
+
+A class declaration, together with its constructor parameters, its `extends` clause, and its self type, is technically a single line.
+Example:
+
+```scala
+class Foo(val x: Int) extends Bar with Foobar { self =>
+```
+
+However, this tends to become too long in many cases.
+
+If the declaration does not fit on one line, the self type must be on dedicated line, indented 2 spaces only, and followed by a blank line:
+
+```scala
+class Foo(val x: Int) extends Bar with Foobar {
+  self =>
+
+  // declarations start here
+```
+
+If too long in itself, the list of constructor parameters should be broken similarly to formal parameters to a method, i.e., indented 4 spaces, and followed by a blank line:
+
+```scala
+class Foo(val x: Int, val y: Int,
+    val z: Int) extends Bar with Foobar {
+
+  // declarations start here
+```
+
+As an exception, if the constructor parameters are a (long) list of "configuration" parameters, the following format should be used instead:
+
+```scala
+class Foo(
+    val width: Int = 1,
+    val height: Int = 1,
+    val depthOfField: Int = 3
+) extends Bar with Foobar {
+```
+
+Note that there is no vertical alignment, neither for the type nor the default value (if any).
+If there are several parameter lists (e.g., with an implicit parameter list), each parameter list follows its rules independently of the others, i.e., organizing one parameter list vertically does not mean another list should be organized vertically as well.
+For example:
+
+```scala
+class Foo[A](
+    val width: Int = 1,
+    val height: Int = 1,
+    val depthOfField: Int = 3
+)(implicit ct: ClassTag[A]) extends Bar with Foobar {
+```
+
+If everything up to the first `with`, excluded, fits on one line, then the list of `with`s should be indented so as to align with `extends`, such as:
+
+```scala
+class Foo(val x: Int) extends Bar
+                         with Foobar
+                         with AnotherTrait {
```

Comment:
Oh we do have it in quite a lot of places.
But I agree, it does not look like the rest of the coding style. I'll change that.